flink-runtime/src/main/java/org/apache/flink/runtime/io/network/partition/PipelinedSubpartition.java

```diff
@@ -350,10 +350,21 @@ BufferAndBacklog pollBuffer() {
     }
 
     void resumeConsumption() {
+        final boolean notifyDataAvailable;
         synchronized (buffers) {
             checkState(isBlocked, "Should be blocked by checkpoint.");
 
             isBlocked = false;
+
+            // When buffer timeout is 0, there might be data produced during
+            // the channel is blocking and do not notify due to blocking, and
+            // it would not get chance to notify, thus we should notify for it
+            // here.
+            notifyDataAvailable = isDataAvailableUnsafe();
+        }
+
+        if (notifyDataAvailable) {
+            notifyDataAvailable();
```

Comment:
For RemoteInputChannel, the available reader should have been enqueued as available when resuming consumption in PartitionRequestQueue#addCreditOrResumeConsumption method, this data available notification can be duplicated. Maybe we can keep the logic of LocalInputChannel#resumeConsumption method unchanged and only fix the flushRequested flag? What do you think?

Reply:
Very thanks pointing out this issue! Sorry that I ignored the logic corresponding to the `RemoteInputChannel`. I think it is reasonable to not introduce the duplicate notification, I'll modify the code.